icepyx/core/query.py

```diff
@@ -886,7 +889,7 @@ def show_custom_options(self, dictview=False):
             all(key in self._cust_options.keys() for key in keys)
         except AttributeError or KeyError:
             self._cust_options = is2ref._get_custom_options(
-                self._session, self.product, self._version
+                self.session, self.product, self._version
```

Comment:
would this now need to be `self.auth.session` (or `self.auth._session`)?

Reply:
All instances of `self.auth._session` should now be `self.auth._session`, and similarly for `self.auth.s3login_credentials`. The private variable can be accessed, but doing so would bypass the automatic setting of the session or s3 credentials.